File: atividade-10/atividade.go
package main

import "fmt"

type Movel interface {
	Mover() string
}

type Som interface {
	EmitirSom() string
}

type Cachorro struct{}

func (c Cachorro) EmitirSom() string {
	return "Latido"
}

type Carro struct{}

func (c Carro) Mover() string {
	return "O carro está se movendo"
}

func main() {
	var som Som = Cachorro{}
	fmt.Println(som.EmitirSom()) 

	var movel Movel = Carro{}
	fmt.Println(movel.Mover()) 
}

File: atividade-10/atividade.py
from abc import ABC, abstractmethod

class Animal(ABC):
    @abstractmethod
    def som(self):
        pass

class Cachorro(Animal):
    def som(self):
        return "Latido"

class Movel(ABC):
    @abstractmethod
    def mover(self):
        pass

class Carro(Movel):
    def mover(self):
        return "O carro está se movendo"

cachorro = Cachorro()
print(cachorro.som())  

carro = Carro()
print(carro.mover())